constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/ShiftBy.java

```diff
@@ -0,0 +1,60 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.ActivityInstance;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Window;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public final class ShiftBy implements Expression<Windows> {
+  public final Expression<Windows> windows;
+  public final Duration fromStart;
+  public final Duration fromEnd;
+
+  public ShiftBy(final Expression<Windows> left, final Duration clampFromStart, final Duration clampFromEnd) {
```

Comment:
:small_red_triangle_down: Should these parameters be renamed to `shiftFromStart` and `shiftFromEnd` ?